src/averageAges.js

```diff
@@ -36,9 +40,21 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
-}
+  const women = people.filter(person => person.sex === 'f');
+  const womenAges = women.map(person => person.died - person.born);
+  const womenWithChildren = women.filter(person =>
+    people.some(human => human.mother === person.name
+      || human.mother === person.name));
+  const womenWithChildrenAge = womenWithChildren.map(person =>
+    person.died - person.born);
+
+  return withChildren
+    ? +((womenWithChildrenAge.reduce((a, b) => a + b, 0)
+    / womenWithChildren.length).toFixed(2))
+    : +((womenAges.reduce((a, b) => a + b, 0) / women.length).toFixed(2));
```

Comment:
You do the same on `calculateMenAverageAge`. You can create a separate function and use it here and in `calculateMenAverageAge`.